src/webgpu/api/validation/shader_module/entry_point.spec.ts

```diff
@@ -93,25 +149,149 @@ Tests calling createRenderPipeline(Async) with valid fragment stage shader and d
 and check that the APIs only accept matching entryPoint.
 `
   )
-  .params(u => u.combine('isAsync', [true, false]).combineWithParams(kEntryPointTestCases))
+  .params(u =>
+    u
+      .combine('isAsync', [true, false])
+      .combine('shaderModuleStage', ['compute', 'vertex', 'fragment'] as const)
+      .beginSubcases()
+      .combine('provideEntryPoint', [true, false])
+      .combine('extraEntryPoint', [true, false])
+      .combineWithParams(kEntryPointTestCases)
+  )
   .fn(t => {
-    const { isAsync, shaderModuleEntryPoint, stageEntryPoint } = t.params;
+    const {
+      isAsync,
+      provideEntryPoint,
+      extraEntryPoint,
+      shaderModuleStage,
+      shaderModuleEntryPoint,
+      stageEntryPoint,
+    } = t.params;
+    const entryPoint = provideEntryPoint ? stageEntryPoint : undefined;
+    let code = getShaderWithEntryPoint(shaderModuleStage, shaderModuleEntryPoint);
+    if (extraEntryPoint) {
+      code += ` ${getShaderWithEntryPoint(shaderModuleStage, 'extra')}`;
+    }
     const descriptor: GPURenderPipelineDescriptor = {
       layout: 'auto',
       vertex: {
         module: t.device.createShaderModule({
           code: kDefaultVertexShaderCode,
         }),
-        entryPoint: 'main',
       },
       fragment: {
         module: t.device.createShaderModule({
-          code: getShaderWithEntryPoint('fragment', shaderModuleEntryPoint),
+          code,
         }),
-        entryPoint: stageEntryPoint,
+        entryPoint,
         targets: [{ format: 'rgba8unorm' }],
       },
     };
-    const _success = shaderModuleEntryPoint === stageEntryPoint;
+    let _success = true;
+    if (shaderModuleStage !== 'fragment') {
+      _success = false;
+    }
+    if (!provideEntryPoint && extraEntryPoint) {
+      _success = false;
+    }
+    if (shaderModuleEntryPoint !== stageEntryPoint && provideEntryPoint) {
+      _success = false;
+    }
     t.doCreateRenderPipelineTest(isAsync, _success, descriptor);
   });
+
+g.test('compute_undefined_entry_point_and_extra_stage')
+  .desc(
+    `
+Tests calling createComputePipeline(Async) with compute stage shader and
+an undefined entryPoint is valid if there's an extra shader stage.
+`
+  )
+  .params(u =>
+    u
+      .combine('isAsync', [true, false])
+      .combine('extraShaderModuleStage', ['vertex', 'fragment'] as const)
+  )
+  .fn(t => {
+    const { isAsync, extraShaderModuleStage } = t.params;
+    const code = `
+        ${getShaderWithEntryPoint('compute', 'main')}
+        ${getShaderWithEntryPoint(extraShaderModuleStage, 'extra')}
+    `;
+    const descriptor: GPUComputePipelineDescriptor = {
+      layout: 'auto',
+      compute: {
+        module: t.device.createShaderModule({
+          code,
+        }),
+        entryPoint: undefined,
+      },
+    };
+    t.doCreateComputePipelineTest(isAsync, true, descriptor);
```

Comment:
For validation tests, there should be some control case where a failure is expected, to make sure the test is doing what we think. In this case it could simply be adding `'compute'` to the `extraShaderModuleStage` list. Similar below.

Reply:
I'll fix this and merge.